scripts/exrtma3d_gsianl.ksh

```diff
@@ -148,60 +153,141 @@ fi
+  ${LN} -s ${DATAOBSHOME}/hrrr.t${HH}${subcyc}z.NASALaRCCloudInGSI.bufr ./larcInGSI
+elif [ -r "${DATAOBSHOME}/rtma_ru.t${HH}${subcyc}z.lgycld.tm00.bufr_d" ]; then
+  ${LN} -s ${DATAOBSHOME}/rtma_ru.t${HH}${subcyc}z.lgycld.tm00.bufr_d ./larcInGSI
+elif [ -r "${DATAOBSHOME}/${RUN}.t${HH}${subcyc}z.NASALaRCCloudInGSI.bufr" ]; then
+  ${LN} -s ${DATAOBSHOME}/${RUN}.t${HH}${subcyc}z.NASALaRCCloudInGSI.bufr ./larcInGSI
 else
   ${ECHO} "Warning: ${DATAOBSHOME}: NASALaRCCloudInGSI.bufr does not exist!"
 fi
 
-# Link statellite radiance data
+# Set runtime and save directories
+export endianness=Big_Endian
+
+# Set variables used in script
+#   ncp is cp replacement, currently keep as /bin/cp
+ncp=/bin/cp
+
+export HYB_ENS=".true."
+
+# Get Fv3GDAS Enkf files
+# We expect 80 total files to be present (80 enkf)
+export nens=80
+
+# Not using FGAT or 4DEnVar, so hardwire nhr_assimilation to 3
+export nhr_assimilation=03
+##typeset -Z2 nhr_assimilation
+
 
-# Link the radial velocity data
+python ${UTILrtma3d_dev}/getbest_EnKF_FV3GDAS.py -v $YYYYMMDDHH --exact=no --minsize=${nens} -d ${COMINGDAS}/enkfgdas -m no -o filelist${nhr_assimilation} --o3fname=gfs_sigf${nhr_assimilation} --gfs_nemsio=yes
+ 
 
-## 
-## Find closest GFS EnKF forecast to analysis time
-##
-## 
-## Link to pre-processed GFS EnKF forecast members
-##
+#Check to see if ensembles were found 
+numfiles=`cat filelist03 | wc -l`
+
+if [ $numfiles -ne 80 ]; then
+  echo "Ensembles not found - turning off ifhyb!"
+  export ifhyb=".false."
+else
+#   we have 80 files, figure out if they are all the right size
+#   if not, set ifhyb=false
+    cp ${UTILrtma3d_dev}/convert.sh .
+    ${UTILrtma3d_dev}/check_enkf_size.sh
+fi
+
+if [ ${HRRRDAS_BEC} -gt 0 ]; then
+  ${ECHO} "\$HRRRDAS_BEC=${HRRRDAS_BEC}, so HRRRDAS will be used if available"
+  #----------------------------------------------------
+  # generate list of HRRRDAS members for ensemble covariances
+  # Use 1-hr forecasts from the HRRRDAS cyclin
+  c=1
+   ${LS} ${COMINhrrrdas}/hrrr.t${HH}00z.f0100.mem000${c}.netcdf > filelist.hrrrdas
+
+  c=2
+  while [[ $c -le 36 ]]; do
+   if [ $c -lt 10 ]; then
+    cc="0"$c
+   else
+    cc=$c
+   fi
+   hrrre_file=${COMINhrrrdas}/hrrr.t${HH}00z.f0100.mem00${cc}.netcdf
+   ${LS} ${COMINhrrrdas}/hrrr.t1000z.f0100.mem00${cc}.netcdf >> filelist.hrrrdas
+   ${LN} -sf ${hrrre_file} wrf_en0${cc}
+   ((c = c + 1))
+  done
+else
+  ${ECHO} "\$HRRRDAS_BEC=${HRRRDAS_BEC}, so HRRRDAS will NOT be used"
+  touch filelist.hrrrdas #so as to avoid "no such file" error message
+fi
 
 # Determine if hybrid option is available
 beta1_inv=1.0
 ifhyb=.false.
+nummem=`more filelist03 | wc -l`
+nummem=$((nummem - 3 ))
+hrrrmem=`more filelist.hrrrdas | wc -l`
+hrrrmem=$((hrrrmem - 3 ))
+if [[ ${hrrrmem} -gt 30 ]] && [[ ${HRRRDAS_BEC} -eq 1  ]]; then #if HRRRDAS BEC is available, use it as first choice
+  echo "Do hybrid with HRRRDAS BEC"
+  nummem=${hrrrmem}
+  cp filelist.hrrrdas filelist03
+
+  beta1_inv=0.50 #0.15
```

Comment:
the parameter "beta1_inv" might be changed for tuning. If it could be modified in xml file, it might be more convenient.  But this is just a suggestion. You do not need to set it up for this version. We can discuss it later.